### DISCUSSION TITLE
My library (Visual Studio project) to VCPKG (private registry)

### DISCUSSION THREAD

**Discussion by yelandgit** [microsoft/vcpkg#36557] [General]
I have C++ libraries created long time ago as Visual Studio project. I'd like to use today this libraries as VCPKG private registry. I've read VCPKG documentation, but I haven't seen the clear answer about package source project type. How I understand for this moment package source must be CMake project, therefore I have to transform my library (Visual Studio project) to CMake project. Is it correct? No other ways?

**Comment by Osyotr** [comment]
There are some libraries that use msbuild instead of cmake.
https://github.com/search?q=repo%3Amicrosoft%2Fvcpkg+vcpkg_msbuild_install&type=code

**Comment by yelandgit** [comment]
Thank you for help

**Comment by dg0yt** [comment]
> How I understand for this moment package source must be CMake project, therefore I have to transform my library (Visual Studio project) to CMake project. I

No. vcpkg supports several build systems incl. msbuild. (However, vcpkg uses CMake as a general portable scripting language.)

**Reply by dg0yt** [reply]
There is the generic example:
https://learn.microsoft.com/en-us/vcpkg/examples/packaging-github-repos
And there are suggested examples:
https://learn.microsoft.com/en-us/vcpkg/examples/packaging-github-repos#suggested-example-portfiles
And then there is documentation for individual functions such as:
https://learn.microsoft.com/en-us/vcpkg/maintainers/functions/vcpkg_build_msbuild

**Comment by yelandgit** [comment]
Ok, great! But where is description in documentation? Where is description how to do it for different build systems?
